src/modules/TurboLens.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.10;
+
+import {ERC20} from "solmate/tokens/ERC20.sol";
+import {ERC4626} from "solmate/mixins/ERC4626.sol";
+
+import {PriceFeed, Comptroller} from "../interfaces/Comptroller.sol";
+import {TurboClerk} from "./TurboClerk.sol";
+import {TurboBooster} from "./TurboBooster.sol";
+
+import {TurboMaster, TurboSafe} from "../TurboMaster.sol";
+
+interface UncompliantERC4626 {
+    function previewRedeem(uint shares) external returns(uint);
+}
+
+/// @title Turbo Lens
+contract TurboLens {
+    Comptroller public immutable pool;
+
+    TurboMaster public immutable master;
+
+    struct StrategyInfo {
+        /// @notice the fei strategy boosted by the safe
+        ERC4626 strategy;
+        
+        /// @notice the amount of fei boosted by the safe to this strategy
+        uint256 boostedAmount;
+        
+        /// @notice the amount of fei held by the safe in this strategy
+        uint256 feiAmount;
+    }
+
+    struct SafeInfo {
+        /// @notice the safe collateral
+        ERC20 collateralAsset;
+
+        /// @notice the balance of collateral assets held by the safe
+        uint256 collateralAmount;
+        
+        /// @notice the collateral value of assets
+        uint256 collateralValue;
+
+        /// @notice the FEI debt held by safe
+        uint256 debtAmount;
+
+        /// @notice the value of FEI deby held by safe
+        uint256 debtValue;
+
+        /// @notice the total FEI boosted value
+        uint256 boostedAmount;
+
+        /// @notice the total amount of FEI boosted by the vault
+        uint256 feiAmount;
+
+        StrategyInfo[] strategyInfo;
+    }
+
+    constructor (TurboMaster _master) {
+        master = _master;
+        pool = _master.pool();
+    }
+
+    function getAllUserSafes(address owner) external returns (SafeInfo[] memory) {
+
+    }
+
+    function getSafeInfo(TurboSafe safe) external returns(SafeInfo memory) {
```

Comment:
Does this function modify state? It feels like it should be a view function.

Reply:
It should be view in theory however some 4626 like the very important compound one will be non-view on preview functions therefore that bubbles up

Reply:
Ok decided this should be view and non-compliant strategies shouldn't be allowed will change

Reply:
still needs to be non-view because of compound but using the view on previewRedeem